File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.py
class Solution(object):
    def resultingString(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []

        def func(ch):
            val = abs(ord(stack[-1]) - ord(ch))
            return val == 1 or val == 25
            
        for ch in s:
            if stack and func(ch):
                stack.pop()
            else:
                stack.append(ch)
                
        return ''.join(stack)

File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.js
/**
 * @param {string} s
 * @return {string}
 */
var resultingString = function(s) {
    const stk = [];
    for (let ch of s) {
        let top = stk[stk.length - 1]
        if (
            stk.length &&
            (Math.abs(ch.charCodeAt(0) - top.charCodeAt(0)) === 1 ||
             Math.abs(ch.charCodeAt(0) - top.charCodeAt(0)) === 25)
        )
            stk.pop()
        else
            stk.push(ch)
    }
    
    return stk.join('');
};